README.md

```diff
@@ -72,12 +72,68 @@ if __name__ == '__main__':
 
 ### CLI Usage
 
-- Lists all datasources in your site
-  - `tableau_utilities --user <login> --password <password> --site <site name> --server <server address> --list_datasources`
-  - OR `tableau_utilities --settings_path ./settings.yaml --list_datasources`
-- Download a datasource
-  - `tableau_utilities --user <login> --password <password> --site <site name> --server <server address> --download_ds --name "Datasource Name" --project "Project Name"`
-  - OR `tableau_utilities --settings_path ./settings.yaml --download_ds --name "Datasource Name" --project "Project Name"`
+#### Help
+1. See the top level CLI arguments including the commands.
+```commandline
+tableau_utilities --help
+```
+
+2. See the help for a `command`.  Each command has arguments for a different grouping of actions.
+```commandline
+tableau_utilities server_operate --help
+```
+
+#### Authentication Options
+1. Pass your credentials into the command
+```commandline
+ tableau_utilities --auth args_token --token_name my_token_name --token_secret 1q2w3e4r5t6y7u8i9o --site mysitename --server 10az server_info --list_object datasource --list_format names
+```
+
+2. Use a secrets yaml file
+```commandline
+ tableau_utilities --auth settings_yaml  --settings_path my_settings.yaml --list_object datasource --list_format names
+```
```

Comment:
What if we just make it so that you need to use a `settings.yaml` file when you use the CLI (dbt style)?
If we did that, I think we could greatly simplify the code and this process in general; i.e. we wouldn't need to specify args for any creds, or the path to the settings file (it would be local to where `tableau_utilities is called).

Reply:
I'd like to keep this approach for these reasons:
1. `settings.yaml` files don't feel as standard as a CLI that just works with OS environment variables.  Being able to use OS variables or passing the auth right into the command feels much more standard to me from other CLIs I've seen.
2. I couldn't get the `settings.yaml` file to work with the 1password cli so that would be a particular block for us.  We don't want to store credentials in plain text local files as that's not a security best practice.  Do you have the settings.yaml file working with a password manager or encryption of some sort.

Reply:
Gotcha. We are not using the CLI in a production environment like that, so we've not had these issues. However, if you have a use-case where having a settings.yaml file would be difficult, I am fine with passing the creds as args.

Reply:
Cool. I'll leave this as is then! Thanks!

And this is set up so you can still authenticate your way too!